Tribler/Core/Libtorrent/LibtorrentDownloadImpl.py

```diff
@@ -706,6 +724,26 @@ def get_current_speed(self, dir):
         with self.dllock:
             return self.curspeeds[dir]
 
+    def save_resume_data(self):
+        """
+        method to save resume data. It will return either :
+        stored resume data, or deferred if it's not available
+        """
```

Comment:
Please, always return a deferred as @lfdversluis said. Having to deal with a function returning different data types randomly doesn't make sense.